public/static/docs/api-reference/open.md

```diff
@@ -0,0 +1,139 @@
+
+[Context manager](https://www.python.org/dev/peps/pep-0343/#context-managers-in-the-standard-library)
+to open a DVC-tracked file <abbr>artifact</abbr> as a
+[file object](https://docs.python.org/3/glossary.html#term-file-object).
+
+## Signature
+
+```py
+open(path, repo=None, rev=None, remote=None, mode="r", encoding=None)
+```
+
+## Description
+
+This function is analogous to the
+[`open()`](https://docs.python.org/3/library/functions.html#open) Python
+builtin, but for files tracked in <abbr>DVC projects</abbr>. However, it may
+only be used as a _context manager_ (using the `with` keyword as shown in the
+examples below). There is no `close()`.
+
+💡 Note that `dvc.api.open` is able to
+[stream](https://docs.python.org/3/library/io.html) the file directly from
+**most**
+[remote types](/doc/command-reference/remote/add#supported-storage-types)
+(local, S3, Azure, GCP, OSS, SSH, HDFS). Otherwise, the file is downloaded
+regularly into a temporary local path before the file object is made available.
+
+> This has similar uses as the `dvc get` and `dvc import` CLI commands.
+
+## Parameters
+
+- **`path`** - used to specify the location of the target artifact within the
+  source project in `repo`, relative to the project's root.
+
+- `repo` - specifies the location of the source DVC project. Both HTTP and SSH
+  protocols are supported for online Git repository URLs (e.g.
+  `[user@]server:project.git`). `repo` can also be a local file system path to
+  an "offline" project. If not supplied, this defaults to the current working
+  directory.
+
+  > A `dvc.exceptions.NotDvcRepoError` is thrown if `repo` is not a valid DVC
+  > project.
+
+- `rev` - (optional)
+  [Git-revision](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
+  (such as a branch name, a tag, or a commit hash). `rev` only has an effect
+  when a URL is supplied as parameter to `repo`. If not supplied, it uses the
+  default Git revision, `HEAD`.
+
+- `remote` - (optional) name of the [DVC remote](/doc/command-reference/remote)
+  to fetch the target artifact from. If not supplied, the default depends on the
+  value of `repo`. The local cache is used when `repo` is the current working
+  directory (default value of `repo`). when `repo` is an external repository
+  URL, the default project remote is used.
+
+  > A `dvc.exceptions.NoRemoteError` is thrown if no `remote` is specified and
+  > the project has no default remote.
+
+- `mode` - (optional) mirrors the namesake parameter in builtin
+  [`open()`](https://docs.python.org/3/library/functions.html#open). Defaults to
+  `"r"` (read).
+
+- `encoding` - (optional) used to decode contents to a string. Mirrors the
+  namesake parameter in builtin `open()`. Defaults to `"utf-8"`.
+
+## Example: process XML file from an external DVC repository
+
+```py
+from xml.dom.minidom import parse
+
+import dvc.api
+
+with dvc.api.open(
+  "get-started/data.xml",
+  "https://github.com/iterative/dataset-registry"
```

Comment:
Let's use explicit `repo=` in all examples. We will probably change that to be keyword only param in the future. This will prevent any confusion with params order.

Reply:
Good idea. Done. (This was the only one missing.)